Source/WebKit/UIProcess/RemoteLayerTree/RemoteLayerTreeInteractionRegionLayers.mm

```diff
@@ -116,40 +116,41 @@ void updateLayersForInteractionRegions(CALayer *layer, RemoteLayerTreeHost& host
-            bool foundInPosition = false;
-            if (region.type == InteractionRegion::Type::Occlusion) {
-                RetainPtr<CALayer> occlusionLayer;
-
-                auto layerIterator = existingOcclusionLayers.find(rect);
-                if (layerIterator != existingOcclusionLayers.end()) {
-                    occlusionLayer = layerIterator->value;
-                    existingOcclusionLayers.remove(layerIterator);
-                    if ([layer.sublayers objectAtIndex:insertionPoint] == occlusionLayer)
-                        foundInPosition = true;
-                    else
-                        [occlusionLayer removeFromSuperlayer];
-                } else {
-                    occlusionLayer = adoptNS([[CALayer alloc] init]);
-                    [occlusionLayer setFrame:rect];
-                    [occlusionLayer setHitTestsAsOpaque:YES];
-                    setInteractionRegionOcclusion(occlusionLayer.get());
-
-                    if (applyBackgroundColorForDebugging) {
-                        [occlusionLayer setBorderColor:cachedCGColor({ WebCore::SRGBA<float>(1, 0, 0, .2) }).get()];
-                        [occlusionLayer setBorderWidth:6];
-                        [occlusionLayer setName:@"Occlusion"];
-                    }
+        IntRect rect = region.rectInLayerCoordinates;
+        bool foundInPosition = false;
+        if (region.type == InteractionRegion::Type::Occlusion) {
+            RetainPtr<CALayer> occlusionLayer;
+
+            auto layerIterator = existingOcclusionLayers.find(rect);
+            if (layerIterator != existingOcclusionLayers.end()) {
+                occlusionLayer = layerIterator->value;
+                existingOcclusionLayers.remove(layerIterator);
+                if ([layer.sublayers objectAtIndex:insertionPoint] == occlusionLayer)
+                    foundInPosition = true;
+                else
+                    [occlusionLayer removeFromSuperlayer];
+            } else {
+                occlusionLayer = adoptNS([[CALayer alloc] init]);
+                [occlusionLayer setFrame:rect];
+                [occlusionLayer setHitTestsAsOpaque:YES];
+                setInteractionRegionOcclusion(occlusionLayer.get());
+                
+                if (applyBackgroundColorForDebugging) {
+                    [occlusionLayer setBorderColor:cachedCGColor({ WebCore::SRGBA<float>(1, 0, 0, .2) }).get()];
+                    [occlusionLayer setBorderWidth:6];
+                    [occlusionLayer setName:@"Occlusion"];
                 }
-
-                if (!foundInPosition)
-                    [layer insertSublayer:occlusionLayer.get() atIndex:insertionPoint];
-
-                insertionPoint++;
-                continue;
             }
+            
+            if (!foundInPosition)
+                [layer insertSublayer:occlusionLayer.get() atIndex:insertionPoint];
 
+            [layer insertSublayer:occlusionLayer.get() atIndex: insertionPoint];
```

Comment:
rebase issue?
we should only insert the occlusion layer in the `if (!foundInPosition)` above.